hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/DataEntity.java

```diff
@@ -141,7 +147,7 @@ static Role valueOf(String name) {
         }
 
         public Blueprint(DataRole role, StructuredData value, Map<String, Object> properties) {
-            super(properties);
+            super(role.name(), properties);
```

Comment:
The constructor is private (it's there because of jackson). Have you seen the NPE from the agent?